build.sbt

```diff
@@ -0,0 +1,154 @@
+val sparkVersion = "2.3.1"
+val catsCoreVersion = "1.4.0"
+val catsEffectVersion = "1.0.0"
+val catsMtlVersion = "0.3.0"
+val shapeless = "2.3.2"
+
+val scalatest = "3.0.3"
+val scalacheck = "1.13.5"
+
+lazy val volalandSettings = Seq(
+  organization := "land.vola",
+  scalaVersion := "2.11.12",
+  scalacOptions ++= commonScalacOptions,
+  licenses += ("Apache-2.0", url("http://opensource.org/licenses/Apache-2.0")),
+  testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oDF"),
+  libraryDependencies ++= Seq(
+    "com.chuusai" %% "shapeless" % shapeless,
+    "org.scalatest" %% "scalatest" % scalatest % "test",
+    "org.scalacheck" %% "scalacheck" % scalacheck % "test"),
+  javaOptions in Test ++= Seq("-Xmx1G", "-ea"),
+  fork in Test := true,
+  parallelExecution in Test := false
+)
+
+lazy val commonScalacOptions = Seq(
+  "-target:jvm-1.8",
+  "-deprecation",
+  "-encoding", "UTF-8",
+  "-feature",
+  "-unchecked",
+  "-Xfatal-warnings",
+  "-Xlint:-missing-interpolator,_",
+  "-Yinline-warnings",
+  "-Yno-adapted-args",
+  "-Ywarn-dead-code",
+  "-Ywarn-numeric-widen",
+  "-Ywarn-value-discard",
+  "-Ywarn-unused-import",
+  "-language:existentials",
+  "-language:experimental.macros",
+  "-language:implicitConversions",
+  "-language:higherKinds",
+  "-Xfuture"
+)
+
+
+lazy val root = Project("volaland", file("." + "volaland")).in(file("."))
+  .aggregate(core, cats, dataset, ml, docs)
+  .settings(volalandSettings: _*)
+
+lazy val core = project
+  .settings(name := "volaland-core")
+  .settings(volalandSettings: _*)
+
+
+lazy val cats = project
+  .settings(name := "volaland-cats")
+  .settings(volalandSettings: _*)
+  .settings(
+    addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.4"),
+    scalacOptions += "-Ypartial-unification"
+  )
+  .settings(libraryDependencies ++= Seq(
+    "org.typelevel"    %% "cats-core"      % catsCoreVersion,
+    "org.typelevel"    %% "cats-effect"    % catsEffectVersion,
+    "org.typelevel"    %% "cats-mtl-core"  % catsMtlVersion,
+    "org.typelevel"    %% "alleycats-core" % catsCoreVersion,
```

Comment:
What about cats.laws and cats.test in Test?

Reply:
now there are minimal dependencies that will be used in some way. 
let's add dependencies in moment when they really used.

Reply:
:+1: